File: generators/c++_mt19937/generator.cpp
#include <iostream>
#include <fstream>
#include <random>


constexpr int BITSTRING_SIZE = 16'000'000;


int main()
{
	std::random_device rd;
	std::mt19937 mt(rd());
	std::uniform_int_distribution<uint32_t> dist(0, UINT32_MAX);

	std::ofstream outfile("./randomness.txt");
	for (int i = 0; i < BITSTRING_SIZE / 4; ++i)
	{
		uint32_t r = dist(mt);
        outfile.write(reinterpret_cast<char*>(&r), sizeof(r));
	}

	outfile.close();
	return 0;
}

File: generators/python_random/generator.py
import random
import struct


BITSTRING_SIZE = 16_000_000


with open("randomness.txt", "wb") as f:
	for _ in range(int(BITSTRING_SIZE / 4)):
		r = random.randint(0, 2**32 - 1)
		b = struct.pack('>I', r)
		f.write(b)

